DIY_List/src/main/java/ru/otus/homework2/DiyLIST.java

```diff
@@ -0,0 +1,248 @@
+package ru.otus.homework2;
+
+import java.util.*;
+
+public class DiyLIST<E> implements List<E> {
```

Comment:
Для классов принято CamelCase
```suggestion
public class DiyList<E> implements List<E> {
```